### PR TITLE
Adds class for upvote view

### PR DESCRIPTION
**Pull request by agbilotia1998**

### CODE REVIEW

File: src/restapi/ideas/views.py [hunk 1]
```diff
@@ -27,24 +27,24 @@ def get_queryset(self):
...
-    serializer = IdeaSerializer(idea, data = {'upvotes': upvotes}, partial = True)
-
-    if serializer.is_valid():
-        serializer.save()
-
-        return Response(serializer.data, status.HTTP_201_CREATED)
-
-    return Response(serializer.errors, status.HTTP_400_BAD_REQUEST) 
+class UpvotesView(views.APIView):
+    def put(self, request, idea_id):
```
Comment (WickedBrat) [inline src/restapi/ideas/views.py]: Please add permission_classes  and define serializer here itself.

File: src/restapi/ideas/views.py [hunk 2]
```diff
@@ -27,24 +27,24 @@ def get_queryset(self):
...
+        is_idea_upvoted = user.upvoted_ideas.filter(idea_id=idea_id)
+        upvotes = idea.upvotes
+
+        if is_idea_upvoted:
+            upvotes -= 1
+            user.upvoted_ideas.filter(idea_id=idea_id).delete()
+        else:
+            upvotes += 1
+            user.upvoted_ideas.create(idea=idea, user=user)   
+        serializer = IdeaSerializer(idea, data = {'upvotes': upvotes}, partial = True)
```
Comment (WickedBrat) [inline src/restapi/ideas/views.py]: This should be defined upwords

File: src/restapi/ideas/views.py (hunk 1 above)
Comment (agbilotia1998) [reply src/restapi/ideas/views.py]: This is not a generic view, default permission class from setting is already applied here.

File: src/restapi/ideas/views.py (hunk 2 above)
Comment (agbilotia1998) [reply src/restapi/ideas/views.py]: Again this is an API view and to update upvotes IdeaSerializer needs to be passed the updated count.

File: src/restapi/ideas/views.py (hunk 1 above)
Comment (WickedBrat) [reply src/restapi/ideas/views.py]: Are you able to make a successful query with this? 
Please pull the code for frontend and try to see if you make a successful request by pressing the like button. The service currently adds the access token automatically.

Comment (agbilotia1998) [reply src/restapi/ideas/views.py]: The API seems to work fine, please mention the endpoint used.

**Comment by WickedBrat** [comment]
@agbilotia1998 Please add the "user_id": bool field in the response to this endpoint. This would be then used to update button

**Pull request closed without merging**